cpp/src/unary/math_ops.cu

```diff
@@ -234,6 +234,21 @@ struct DeviceNot {
   }
 };
 
+// negation
+
+struct DeviceNegate {
+  template <typename T, CUDF_ENABLE_IF(std::is_signed_v<T> || cudf::is_duration_t<T>::value)>
+  T __device__ operator()(T data)
+  {
+    return -data;
+  }
+  template <typename T, CUDF_ENABLE_IF(!std::is_signed_v<T> && !cudf::is_duration_t<T>::value)>
+  T __device__ operator()(T data)
+  {
+    return data;
+  }
```

Comment:
I am confused by this implementation. I thought we didn't want to provide an implementation for unsigned data types?

Reply:
I think that needs to happen at the dispatch layer, not SFINAE on the device operator itself. For example, we're not constraining types in this snippet:

```cpp
struct DeviceSqrt {
  template <typename T>
  __device__ T operator()(T data)
  {
    return std::sqrt(data);
  }
};
```

Can we remove the SFINAE and just have this?
```cpp
struct DeviceNegate {
  template <typename T>
  T __device__ operator()(T data)
  {
    return -data;
  }
};
```

Also related to https://github.com/rapidsai/cudf/pull/17560/files#r1878099060

Reply:
Let me post the compile error I got when I removed it.